rasterio/mask.py

```diff
@@ -3,132 +3,182 @@
 import logging
 import warnings
 
-import rasterio
-from rasterio.features import geometry_mask
+import numpy as np
+
+from rasterio.features import geometry_mask, geometry_window
 
 
 logger = logging.getLogger(__name__)
 
 
-def mask(raster, shapes, nodata=None, crop=False, all_touched=False,
-         invert=False, pad=False):
-    """Mask the area outside of the input shapes with nodata.
+def raster_geom_mask(raster, shapes, all_touched=False, invert=False,
```

Comment:
Function names elsewhere are writing out `geometry` rather than the abbreviation `geom`. I know that's a little verbose here, but seems preferable to me.